### PR TITLE
[Hexagon] Fix loop carry type

### PR DESCRIPTION
**Pull request by dsharletg**
@abadams please take a look

### CODE REVIEW

File: src/LoopCarry.cpp [hunk 1]
```diff
@@ -491,8 +490,8 @@ class LoopCarryOverLoop : public IRMutator {
             }
 
             allocs.push_back({scratch,
-                        loads[0][0]->type.element_of(),
-                        (int)c.size() * loads[0][0]->type.lanes(),
+                        loads[c.front()][0]->type.element_of(),
```
Comment (ronlieb) [inline src/LoopCarry.cpp]: The complexity of a problem is inversely proportional to the fix when measured in keystrokes :-).

**Comment by ronlieb** [comment]
LGTM, working for me here...

**Comment by abadams** [comment]
lgtm. Sorry for the original bug.

**Comment by dsharletg** [comment]
No problem! Just wanted to make sure you thought the fix was correct.

**Pull request merged**